strapi/strapi
# fix: admin build path when serverAdminPanel is false

### What does it do?

determine `basePath` based on the `serveAdminPanel` value

### Why is it needed?

By default, `serverAdminPanel` is true and admin panel is served by Strapi back end at `/admin`, therefore the assets are loaded from `/admin/runtime~main.js`

when `serverAdminPanel` is false, it generates relative path (`runtime~main.js`) to support deploying the admin panel separately from the backend.

### How to test it?

-  create a new strapi app with ~~`npx create-strapi-app@0.0.0-experimental.d10cfc2d4947d3dcbf166b29dc168cb4a7123772`~~ ~~`npx create-strapi-app@0.0.0-experimental.4184db001fb3249acae738f57aab40d83602c0e9`~~ `npx create-strapi-app@0.0.0-experimental.6e4965146398480201de1e5fa870efbead50da6a`
-   add `url: "http://blahblah.com"`,  to `./config/server.js` 
-  add `url: "/",` and `serverAdminPanel: false` to `./config/admin.js`([more info here](https://docs-v4.strapi.io/dev-docs/admin-panel-customization#different-servers))
- run `yarn build`
- the `index.html` file located in the `build` folder should have relative path like `src="runtime~main.xxxx.js"` instead of `src="//runtime~main.2e50647e.js"` reported in the issue

### Related issue(s)/PR(s)

fixes #19330

DX-1655
